src/cinder/block_store.go

```diff
@@ -226,16 +240,21 @@ func (b *BlockStore) GetVolumeID(unstructuredPV runtime.Unstructured) (string, e
 		return "", fmt.Errorf("failed to convert from unstructured PV: %w", err)
 	}
 
-	var volumeID string
 	if pv.Spec.Cinder != nil {
-		volumeID = pv.Spec.Cinder.VolumeID
-	} else if pv.Spec.CSI.Driver == "cinder.csi.openstack.org" || pv.Spec.CSI.Driver == "disk.csi.everest.io" {
-		volumeID = pv.Spec.CSI.VolumeHandle
-	} else {
-		return "", fmt.Errorf("persistent volume is missing 'spec.cinder.volumeID' or PV driver ('spec.csi.driver') doesn't match supported drivers(cinder.csi.openstack.org, disk.csi.everest.io)")
+		return pv.Spec.Cinder.VolumeID, nil
 	}
 
-	return volumeID, nil
+	if pv.Spec.CSI == nil {
+		return "", nil
+	}
+
+	if pv.Spec.CSI.Driver == "cinder.csi.openstack.org" || pv.Spec.CSI.Driver == "disk.csi.everest.io" {
+		return pv.Spec.CSI.VolumeHandle, nil
+	}
+
+	b.log.Infof("Unable to handle CSI driver: %s", pv.Spec.CSI.Driver)
```

Comment:
I explicitly avoided warn here since `velero backup` status was finished with warnings in case you're using both Cinder and Manila PVs. This status confuses end users.

I also noticed that this approach is used in official AWS velero plugin, see:
* https://github.com/vmware-tanzu/velero-plugin-for-aws/blob/109ba05302ff24dc61ec1b33121a0a6fff1b3b86/velero-plugin-for-aws/volume_snapshotter.go#L286
* https://github.com/vmware-tanzu/velero-plugin-for-aws/blob/109ba05302ff24dc61ec1b33121a0a6fff1b3b86/velero-plugin-for-aws/volume_snapshotter.go#L296

Reply:
Ok, clear